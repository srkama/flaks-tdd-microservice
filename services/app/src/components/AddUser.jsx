import React from 'react'

export default function AddUser(props) {
    console.log(props);
    const { username, email, handleChange, handleSubmit } = props;
    return (
        <form onSubmit={handleSubmit}>
            <div className="field">
                <label className="label">Name</label>
                <div className="control">
                    <input
                        name="username"
                        className="input"
                        type="text"
                        placeholder="Name"
                        value={username}
                        required
                        onChange={handleChange} />
                </div>
            </div>
            <div className="field">
                <label className="label">Email</label>
                <div className="control">
                    <input
                        name="email"
                        className="input"
                        type="text"
                        placeholder="email"
                        value={email}
                        required
                        onChange={handleChange} />
                </div>
            </div>
            <div className="field is-grouped">
                <div className="control">
                    <input type="submit" name="submit" className="button is-link" value="Submit" />
                </div>
            </div>
        </form >

    )
}
