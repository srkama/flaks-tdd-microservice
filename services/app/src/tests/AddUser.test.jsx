import React from 'react'
import { shallow } from 'enzyme';
import AddUser from '../components/AddUser';
import renderer from 'react-test-renderer';

test('AddUser renders proprely', () => {
    const wrapper = shallow(<AddUser />);
    const formElement = wrapper.find('form');
    const inputElements = formElement.find('input');
    expect(inputElements.length).toBe(3);
    expect(inputElements.get(0).props.name).toBe('username');
    expect(inputElements.get(1).props.name).toBe('email');
    expect(inputElements.get(2).props.name).toBe('submit');
})

test('Add User snapshots renders properly', () => {
    const tree = renderer.create(<AddUser />).toJSON();
    expect(tree).toMatchSnapshot();
})
