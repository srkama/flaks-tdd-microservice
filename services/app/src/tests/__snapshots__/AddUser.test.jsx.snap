// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add User snapshots renders properly 1`] = `
<form>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Name
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        name="username"
        placeholder="Name"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Email
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        name="email"
        placeholder="email"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="field is-grouped"
  >
    <div
      className="control"
    >
      <input
        className="button is-link"
        name="submit"
        type="submit"
        value="Submit"
      />
    </div>
  </div>
</form>
`;
